### ISSUE TITLE
Git refresh should not show hidden submodules

### ISSUE THREAD

**Issue by xmedeko** [microsoft/vscode#88670]
I have manual Git refresh (unchecked Git Autorefresh). I have hid some submodules also (Source Control -> right click on submodule -> Hide). When I invoke command "Git: Refresh" a list of all submodules is shown. I do not want to have hidden submodules in the list.

Note: the hidden submodules are kind of read-only for me and very rarely updated. I can refresh them from a command line if I need to, or show them, refresh and hide them. Maybe some command "Git: Refresh All" could be handy to refresh all submodules at once.

Note 2: have not tested the behaviour with Git Autorefresh, AFAIK hidden submodules should not be auto updated, too.

**Comment by joaomoreno** [comment]
@xmedeko Can you share a screen recording of the issue? Thanks!

**Comment by xmedeko** [comment]
I've made an example `vscodegitbug`. First, I hide the submodules `dojo` and `dijit` in the Source Control bar.

![vscodegitbug_1](https://user-images.githubusercontent.com/1101801/72892060-e98a8b80-3d15-11ea-82d5-18d1ff5618f4.png)

Then, I enter the command "Git: Refresh", but still see all submodules.

![vscodegitbug_2](https://user-images.githubusercontent.com/1101801/72892075-ee4f3f80-3d15-11ea-85b6-a428407b5c14.png)

I would like to see the original module `vscodegitbug` and the `eemitter` module only. (Do not know, why the `vscodegitbug` is not the first in the list).

Maybe some people want to see the hidden submodules, too? Then, I would welcome some way how to hide certain (user defined) submodules completely from the VS Code GUI. Even from the SOURCE CONTROL PROVIDERS list.

**Comment by joaomoreno** [comment]
Oh, when you hide a repository, you don't really close it. Have you tried `Git: Close Repository`? There's also the `git.ignoredRepositories` setting.

**Comment by xmedeko** [comment]
I do not want to close it, I just want to hide it (almost) everywhere in VS Code GUI, not just in the Source Control bar. We have many submodules in the project and i would like to see just a few of them. And I think the `git.ignoredRepositories` does not work this way, does it? I've tried Workspace settings:
```json
    "git.ignoredRepositories": [
        "./src/dojo"
    ]
```
and the repo is still in the "Git: Refresh" list.

**Comment by xmedeko** [comment]
Yes, closing the repository is probably what I need, thanks. I cannot find a doc for that. Does it any git changes action or is it just a pure VS Code action?

Also, is `git.gnoredRepositories` documented somewhere? Thanks.

Note: I've read https://code.visualstudio.com/docs/editor/versioncontrol and tried to google for these features.

**Comment by joaomoreno** [comment]
It's a pure VS Code action.

Since `git.ignoredRepositories` is an extension (albeit built-in) setting, it's not documented. But our settings editor has an excellent search feature built in, you could certainly find it there.

Glad to know this addresses your pain point, closing the issue.

**Comment by xmedeko** [comment]
Yes, i have found the "Git: Ignored Repositories" in the Settings, but the doc "List of git repositories to ignore." is not much of help. How is that different to closing repo?

How to open the repository again? When I try "Git: Open Repository", and choose the submodule path, then I do not see submodule in the Source Control bar. Does it work for submodules?

**Comment by xmedeko** [comment]
OK, found the problem. 
1. First I have set `./src/dojo` into the "Git: Ignored Repositories". But the repository remained in the Source Control Providers bar. 
2. So I have closed it. It disappeared from the Source Control Providers bar. 
3. I have tried to "Git: Open Repository" the repository path, but no luck.
4. I have cleared the "Git: Ignored Repositories" list.
5. Then have tried "Git: Open Repository" and voilà, the repository is in the  Source Control Providers bar again.

It's kind of confusing.

One more though: the reopned repo is the last one in the Source Control Providers, it's ptty it's not sorted alphabetically or possible to reorder.
